com.arc.embeddedcdt/src/com/arc/embeddedcdt/common/DebuggerPluginVersionsChecker.java

```diff
@@ -0,0 +1,61 @@
+package com.arc.embeddedcdt.common;
+
+import java.util.HashSet;
+
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.widgets.Display;
+
+import com.arc.embeddedcdt.LaunchConfigurationConstants;
+import com.arc.embeddedcdt.dsf.utils.Configuration;
+
+public class DebuggerPluginVersionsChecker {
+
+    private HashSet<String> seenLaunchConfigurationNames = new HashSet<>();
+
+    private static DebuggerPluginVersionsChecker INSTANCE = null;
+
+    private ILaunchConfiguration launchConfiguration;
+
+    private DebuggerPluginVersionsChecker() {
+    }
+
+    public static DebuggerPluginVersionsChecker getDebuggerPluginVersionsChecker() {
+        if (INSTANCE == null) {
+            INSTANCE = new DebuggerPluginVersionsChecker();
+        }
+        return INSTANCE;
+    }
+
+    public void checkPluginVersion(ILaunchConfiguration launchConfiguration) {
+        String launchConfigurationName = Configuration.getProgramName(launchConfiguration);
+        if (!seenLaunchConfigurationNames.contains(launchConfigurationName)) {
+            seenLaunchConfigurationNames.add(launchConfigurationName);
+            final int currentLaunchConfigurationDebuggerPluginVersion = Configuration
+                    .getDebuggerPluginVersion(launchConfiguration);
+            if (currentLaunchConfigurationDebuggerPluginVersion != LaunchConfigurationConstants.UNREAL_DEBUGGER_PLUGIN_VERSION_NUMBER
+                    && currentLaunchConfigurationDebuggerPluginVersion != LaunchConfigurationConstants.DEBUGGER_PLUGIN_VERSION_NUMBER) {
+                // Otherwise, a debugger plug-in version was not written to configuration' .launch
+                // file.
+                showWarningWindow(String.format(
+                        "Your debugger plug-in version is %d, but the launch configuration was created with the"
+                                + " plug-in version %d.\n",
+                        LaunchConfigurationConstants.DEBUGGER_PLUGIN_VERSION_NUMBER,
+                        currentLaunchConfigurationDebuggerPluginVersion));
+            }
+        }
+    }
+
+    private void showWarningWindow(final String warningMsg) {
```

Comment:
That wouldn't work on the console builds without GUI. Could you check other cases in CDT of warning windows - is there any standard way to handle, so that if this is a console build, then print warning into stderr?